File: context_tools/stories/templates/py/{epic-verb-noun}/{sub-epic-verb-noun}/{story-verb-noun}/{story_verb_noun}_story.py
# ---
# fidelity: [exploration, specification]
# artifact: [story-file]
# format: py
# section: leaf-story-file
# ---
#
# Conceptual story reference. Refer to context_tools/language-tools.md for tooling.
#
#   Folder tree   -> {epic-verb-noun}/{sub-epic-verb-noun}/{story-verb-noun}/
#   Story file    -> {story_verb_noun}_story.py

"""Story: {Story Name} (conceptual reference)."""

from __future__ import annotations
from typing import Protocol


class {EpicVerbNoun}Helper(Protocol):
    def given_precondition(self) -> None: ...
    def when_action(self) -> None: ...
    def then_outcome(self) -> None: ...


def create_{story_verb_noun}_story(h: "{EpicVerbNoun}Helper") -> dict:
    """Build one pytest test function per scenario.
    """
    tests = {}

    def test_main_flow() -> None:
        """SCENARIO: {main-flow outcome}"""
        h.given_precondition()
        h.when_action()
        h.then_outcome()

    tests["test_main_flow"] = test_main_flow
    return tests

File: context_tools/stories/templates/ts/{epic-verb-noun}/{sub-epic-verb-noun}/{story-verb-noun}/{story_verb_noun}_story.ts
/**
 * Story: {Story Verb-Noun} (conceptual reference).
 * 
 * Refer to context_tools/language-tools.md for tool recommendations.
 * 
 * Tiers: {story_verb_noun}.{tier}.ts implements the conceptual GWT steps
 * (tier ∈ front-end | back-end | external-system).
 */

import { scenario, story } from "../../../story-test";

export interface {StoryVerbNoun}Helper {
  givenPrecondition(): void | Promise<void>;
  whenAction(): void | Promise<void>;
  thenOutcome(): void | Promise<void>;
}

export function create{StoryVerbNoun}Story(h: {StoryVerbNoun}Helper): void {
  story("{Story Verb-Noun}", () => {
    scenario("{main-flow outcome}", ({ given, when, then }) => {
      given("{given step text}", () => h.givenPrecondition());
      when("{when step text}", () => h.whenAction());
      then("{then step text}", () => h.thenOutcome());
    });
  });
}
